modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataMediator.java

```diff
@@ -88,6 +93,15 @@ public MetadataMediator(T componentModel) {
     this.keyIdObjectResolver = new MetadataKeyIdObjectResolver(component);
     this.outputDelegate = new MetadataOutputDelegate(componentModel);
     this.inputDelegate = new MetadataInputDelegate(componentModel);
+    if (componentModel instanceof SourceModel) {
+      successCallbackInputDelegate = ((SourceModel) componentModel).getSuccessCallback()
+          .map(successCallbackModel -> new MetadataInputDelegate(successCallbackModel));
+      errorCallbackInputDelegate = ((SourceModel) componentModel).getErrorCallback()
+          .map(errorCallbackModel -> new MetadataInputDelegate(errorCallbackModel));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this lambda with a method reference. ([squid:S1612](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1612))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7668&issues=AWjdfEMLotIXnmsUGfbY&open=AWjdfEMLotIXnmsUGfbY)

[ISSUE_KEY]: AWjdfEMLotIXnmsUGfbY